app/workers/set_tenant_id_worker.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class SetTenantIdWorker < ApplicationJob
+
+  def perform(*)
+    # Update the providers in batches of 100
+    loop do
+      ids = Account.where("provider = 1 AND master IS NULL").limit(100).to_a.map { |account| account.id }
```

Comment:
You can use `Account.providers.not_master.find_each`
ActiveRecord::Query has handy magic methods ;)
Also `find_each` will use `find_in_batches`

Reply:
I updated the PR following your suggestion, with only one difference: instead of enqueing a separate job for every single provider, I enqueue a job for batches of providers. I think the result would be the same but it wouldn't congest the queue as much